network/router.go

```diff
@@ -346,3 +358,10 @@ func (r *Router) receiveServerIdentity(c Conn) (*ServerIdentity, error) {
 	log.Lvl4(r.address, "Identity received from", dst.Address)
 	return dst, nil
 }
+
+// AddErrorHandler adds a network error handler function for this router. The functions will be called
+// on network error (e.g. Timeout, Connection Closed, or EOF) with the identity of the faulty
+// remote host as 1st parameter.
+func (r *Router) AddErrorHandler(errorHandler func(*ServerIdentity)) {
```

Comment:
A use case that comes into my mind is tests. But in fact you generate a new `Router` for each test anyway, so no, it's not needed. But I'll add it to your issue.